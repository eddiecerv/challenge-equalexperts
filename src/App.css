.App {
  text-align: center;
}

#calculator__body {
  width: 350px;
  margin: auto;
}

.app__bg {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

footer {
  margin-top: 50px;
}

footer p {
  line-height: 1;
  text-align: center;
  font-size: 18px;
}

.calculator__display {
  font-family: 'ZCOOL QingKe HuangYou', cursive;
}