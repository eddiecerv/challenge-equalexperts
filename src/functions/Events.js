const BASIC_OPERATORS = ['*', '/', '+', '-'];
export const INFINITE_DIVIDE_INTO_ZERO_MESSAGE = 'You cannot divide into 0';

export const operateNumbers = (numberA, numberB, type) => {
    numberA = parseFloat(numberA);
    numberB = parseFloat(numberB);
    switch(type) {
        case '-':
            return numberA - numberB;
        case '/':
            if( numberB > 0 ){
                return numberA / numberB;
            } else {
                return INFINITE_DIVIDE_INTO_ZERO_MESSAGE;
            }
        case '*':
            return numberA * numberB;
        default:
            return numberA + numberB;
    }
}

export const doOperation = (value) => {
  let newValue = value;
  BASIC_OPERATORS.forEach( operator => {
    const values = value.split(operator);
    if( values.length > 1 && !isNaN(values[0]) && !isNaN(values[1])) {
        newValue = operateNumbers(values[0], values[1], operator).toString();
    }
  })

  return newValue;
}