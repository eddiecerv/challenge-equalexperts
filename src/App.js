import './App.css';
import Header from './shared/core/Header';
import Main from './components/Main';

function App() {
  return (
    <div className="app__bg">
      <Header />
      <div className='container'>
        <div id='calculator__body'>
          <Main />
        </div>
      </div>
    </div>
  );
}

export default App;
