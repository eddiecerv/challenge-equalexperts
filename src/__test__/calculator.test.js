import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import App from "../App";
import { INFINITE_DIVIDE_INTO_ZERO_MESSAGE, operateNumbers } from '../functions/Events';

const setup = () => render(<App />);

describe('Basic testing for a calculator', () => {
  beforeEach(() => {
    act(() => {
      setup();
    })
  });

  test('Render without crash', () => {  
    const element = screen.getByTestId('title')
    expect(element.innerHTML).toEqual('A simple calculator')
  })

  test('Verify if number buttons are displayed', () => {
    const elements = screen.getAllByRole('button')
    const filteredNumberElements = elements.filter(el => !isNaN(el.innerHTML) ).map( el => parseInt(el.innerHTML) );

    for(let i = 0; i < 10; i++) {
      expect( filteredNumberElements.includes(i) ).toBe(true);
    }
    
  })

  test('Verify basic operations with an event function', () => {
    const sum = operateNumbers('5', '6', '+');
    expect(sum).toEqual(11);

    const sub = operateNumbers('5', '6', '-');
    expect(sub).toEqual(-1);

    const mul = operateNumbers('5', '6', '*');
    expect(mul).toEqual(30);

    const div = operateNumbers('5', '6', '/');
    expect(div.toFixed(2)).toEqual("0.83");
  })

  test('Try to divide into zero', () => {
    const div = operateNumbers('5', '0', '/');
    expect(div).toEqual(INFINITE_DIVIDE_INTO_ZERO_MESSAGE);
  })

  test('Try to put multiple operator symbols', () => {
    const elements = screen.getAllByRole('button')
    const otherElements = ['C', 'CE', '.'];
    const numberButtons = elements.filter(el => !isNaN(el.innerHTML) );
    const operatorButtons = elements.filter(el => isNaN(el.innerHTML) && !otherElements.includes(el.innerHTML) );

    act(() => {
      numberButtons[0].click();
      operatorButtons[0].click();
      operatorButtons[1].click();
      operatorButtons[2].click();
    })
    
    const calculatedValue = screen.getByTestId('displayText');
    const calculatedValueText = calculatedValue.textContent;
    
    const charBeforeOperator = calculatedValueText[calculatedValueText.length - 2];
    const lastChar = calculatedValueText[calculatedValueText.length];
    expect( !isNaN(charBeforeOperator) ).toBe(true);
    expect( isNaN(lastChar) ).toBe(true);
  })

  test('Clean screen after click on C', () => {
    const elements = screen.getAllByRole('button')
    const otherElements = ['C', 'CE', '.'];
    const numberButtons = elements.filter(el => !isNaN(el.innerHTML) );
    const operatorButtons = elements.filter(el => isNaN(el.innerHTML) && !otherElements.includes(el.innerHTML) );
    const equalOperator = elements.find(el => el.innerHTML === '=');
    const cleanButton = elements.find(el => el.innerHTML === 'C');

    act(() => {
      //Make an operation
      numberButtons[5].click();
      operatorButtons[0].click();
      numberButtons[2].click();
      operatorButtons[1].click();
      numberButtons[7].click();
      equalOperator.click();
    })
    
    let calculatedValue = screen.getByTestId('displayText');
    let calculatedValueText = calculatedValue.textContent;
    expect( !isNaN(calculatedValueText) ).toBe(true);
    
    act(() => {
      //Clean screen
      cleanButton.click();
    })

    calculatedValue = screen.getByTestId('displayText');
    calculatedValueText = calculatedValue.textContent;
    expect( calculatedValueText ).toEqual("");
  })
})