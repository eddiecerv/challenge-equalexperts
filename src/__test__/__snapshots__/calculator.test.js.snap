// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic testing for a calculator Generate snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app__bg"
      >
        <header
          class="App-header"
        >
          <p
            data-testid="title"
          >
            A simple calculator
          </p>
        </header>
        <div
          class="container"
        >
          <div
            id="calculator__body"
          >
            <main
              class="card shadow"
            >
              <div
                class="card-body bg-secondary"
              >
                <div
                  class="calculator__display"
                >
                  <div
                    class="shadow-sm"
                    style="padding: 10px; text-align: right; background-color: rgba(66, 135, 245, 0.6); margin-bottom: 10px; min-height: 65px;"
                  >
                    <span
                      data-testid="displayText"
                      style="font-size: 24px;"
                    />
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-lg-3"
                  >
                    <button
                      class="btn btn-outline-light"
                      style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                    >
                      C
                    </button>
                  </div>
                  <div
                    class="col-lg-3"
                  >
                    <button
                      class="btn btn-outline-light"
                      style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                    >
                      CE
                    </button>
                  </div>
                  <div
                    class="col-lg-3"
                  />
                  <div
                    class="col-lg-3"
                  >
                    <button
                      class="btn btn-outline-light"
                      style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                    >
                      /
                    </button>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <section
                    class="col-9"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          7
                        </button>
                      </div>
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          8
                        </button>
                      </div>
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          9
                        </button>
                      </div>
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          4
                        </button>
                      </div>
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          5
                        </button>
                      </div>
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          6
                        </button>
                      </div>
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          1
                        </button>
                      </div>
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          2
                        </button>
                      </div>
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          3
                        </button>
                      </div>
                      <div
                        class="col-lg-4"
                      />
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          0
                        </button>
                      </div>
                      <div
                        class="col-lg-4"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          .
                        </button>
                      </div>
                    </div>
                  </section>
                  <aside
                    class="col-3"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="col-lg-12"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          *
                        </button>
                      </div>
                      <div
                        class="col-lg-12"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          -
                        </button>
                      </div>
                      <div
                        class="col-lg-12"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          +
                        </button>
                      </div>
                      <div
                        class="col-lg-12"
                      >
                        <button
                          class="btn btn-outline-light"
                          style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                        >
                          =
                        </button>
                      </div>
                    </div>
                  </aside>
                </div>
              </div>
            </main>
          </div>
        </div>
        <footer
          class="text-warning"
        >
          <p>
            <small>
              Release Version 
              <br />
              <span
                class="text-secondary"
              >
                v. 5b8d0fd276b6d288905ed2f63a934e057e8feca2
              </span>
            </small>
          </p>
          <p>
            By Eduardo Vallecillo Cervantes
          </p>
        </footer>
      </div>
    </div>
    <div>
      <div
        class="card-body bg-secondary"
      >
        <div
          class="calculator__display"
        >
          <div
            class="shadow-sm"
            style="padding: 10px; text-align: right; background-color: rgba(66, 135, 245, 0.6); margin-bottom: 10px; min-height: 65px;"
          >
            <span
              data-testid="displayText"
              style="font-size: 24px;"
            />
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-lg-3"
          >
            <button
              class="btn btn-outline-light"
              style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
            >
              C
            </button>
          </div>
          <div
            class="col-lg-3"
          >
            <button
              class="btn btn-outline-light"
              style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
            >
              CE
            </button>
          </div>
          <div
            class="col-lg-3"
          />
          <div
            class="col-lg-3"
          >
            <button
              class="btn btn-outline-light"
              style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
            >
              /
            </button>
          </div>
        </div>
        <div
          class="row"
        >
          <section
            class="col-9"
          >
            <div
              class="row"
            >
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  7
                </button>
              </div>
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  8
                </button>
              </div>
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  9
                </button>
              </div>
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  4
                </button>
              </div>
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  5
                </button>
              </div>
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  6
                </button>
              </div>
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  1
                </button>
              </div>
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  2
                </button>
              </div>
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  3
                </button>
              </div>
              <div
                class="col-lg-4"
              />
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  0
                </button>
              </div>
              <div
                class="col-lg-4"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  .
                </button>
              </div>
            </div>
          </section>
          <aside
            class="col-3"
          >
            <div
              class="row"
            >
              <div
                class="col-lg-12"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  *
                </button>
              </div>
              <div
                class="col-lg-12"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  -
                </button>
              </div>
              <div
                class="col-lg-12"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  +
                </button>
              </div>
              <div
                class="col-lg-12"
              >
                <button
                  class="btn btn-outline-light"
                  style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
                >
                  =
                </button>
              </div>
            </div>
          </aside>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-body bg-secondary"
    >
      <div
        class="calculator__display"
      >
        <div
          class="shadow-sm"
          style="padding: 10px; text-align: right; background-color: rgba(66, 135, 245, 0.6); margin-bottom: 10px; min-height: 65px;"
        >
          <span
            data-testid="displayText"
            style="font-size: 24px;"
          />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-3"
        >
          <button
            class="btn btn-outline-light"
            style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
          >
            C
          </button>
        </div>
        <div
          class="col-lg-3"
        >
          <button
            class="btn btn-outline-light"
            style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
          >
            CE
          </button>
        </div>
        <div
          class="col-lg-3"
        />
        <div
          class="col-lg-3"
        >
          <button
            class="btn btn-outline-light"
            style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
          >
            /
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <section
          class="col-9"
        >
          <div
            class="row"
          >
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                7
              </button>
            </div>
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                8
              </button>
            </div>
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                9
              </button>
            </div>
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                4
              </button>
            </div>
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                5
              </button>
            </div>
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                6
              </button>
            </div>
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                1
              </button>
            </div>
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                2
              </button>
            </div>
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                3
              </button>
            </div>
            <div
              class="col-lg-4"
            />
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                0
              </button>
            </div>
            <div
              class="col-lg-4"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                .
              </button>
            </div>
          </div>
        </section>
        <aside
          class="col-3"
        >
          <div
            class="row"
          >
            <div
              class="col-lg-12"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                *
              </button>
            </div>
            <div
              class="col-lg-12"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                -
              </button>
            </div>
            <div
              class="col-lg-12"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                +
              </button>
            </div>
            <div
              class="col-lg-12"
            >
              <button
                class="btn btn-outline-light"
                style="padding: 12px 20px 12px 20px; text-align: center; margin-bottom: 5px; margin-top: 5px; width: 100%;"
              >
                =
              </button>
            </div>
          </div>
        </aside>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
